### PR TITLE
Lesson 10

### PR DESCRIPTION
**Pull request by ValAntG**

### CODE REVIEW

File: app/assets/javascripts/answers.js [hunk 1]
```diff
@@ -1,42 +1,51 @@
 $(document).on("turbolinks:load", function () {
   $('form.new_answer')
-  .bind('ajax:success', function (evt) {
-    if (evt.detail[0].DOCUMENT_TYPE_NODE == 10) {
-      $('.answers-show-form').html(evt.detail[2].responseText);
-    } else {
-      answer = $.parseJSON(evt.detail[2].responseText);
-      $('.answers-show-form').append('<div class="form-horizontal edit-answer-link"></div>');
+    .bind('ajax:success', function (evt) {
```
Comment (AtomGR) [inline app/assets/javascripts/answers.js]: use .on instead of bind

File: app/assets/javascripts/answers.js [hunk 2]
```diff
@@ -1,42 +1,51 @@
...
-      $('.answers-show-form').find('.form-horizontal').find('.form-group').last()
-        .append('<div class="col-sm-offset-3 col-sm-9 attachment"></div>');
-      $('.answers-show-form').find('.form-horizontal').find('.form-group').find('.col-sm-9').last()
-        .append("<a href='" + answer.attachments[0].file.url + "'>" + answer.attachments[0].file.url + '</a>');
+        if (answer.answer.attachments.files) {
+          $('.form-horizontal').last().append('<div class="form-group"></div>');
+          $('.answers-show-form').find('.form-horizontal').find('.form-group').last()
+            .append('<div class="col-sm-3"></div>');
+          $('.answers-show-form').find('.form-horizontal').find('.form-group').find('.col-sm-3').last()
+            .append('<label class="form-control control-label label-warning">Attachments:</label>');
```
Comment (AtomGR) [inline app/assets/javascripts/answers.js]: TRY

File: app/assets/javascripts/answers.js [hunk 3]
```diff
@@ -1,42 +1,51 @@
...
-      $('.answers-show-form').find('.form-horizontal').find('.form-group').find('.col-sm-3').last()
-        .append('<label class="form-control control-label label-warning">Ответ:</label>');
-      $('.answers-show-form').find('.form-horizontal').find('.form-group').last().append('<div class="col-sm-9"></div>');
-      $('.answers-show-form').find('.form-horizontal').find('.form-group').find('.col-sm-9').last()
-        .append('<div class="form-control">' + answer.answer.body + '</div>');
+        $('.form-horizontal').last().append('<div class="form-group"></div>');
+        $('.answers-show-form').find('.form-horizontal').find('.form-group').last()
+          .append('<div class="col-sm-3"></div>');
+        $('.answers-show-form').find('.form-horizontal').find('.form-group').find('.col-sm-3').last()
+          .append('<label class="form-control control-label label-warning">Ответ:</label>');
```
Comment (AtomGR) [inline app/assets/javascripts/answers.js]: TRY

File: app/forms/answer_form.rb [hunk 4]
```diff
@@ -1,6 +1,6 @@
 class AnswerForm
```
Comment (svidlonix) [inline app/forms/answer_form.rb]: What profit do you get when using it?

File: spec/controllers/comments_controller_spec.rb [hunk 5]
```diff
@@ -0,0 +1,212 @@
+require 'rails_helper'
+
+RSpec.describe CommentsController, type: :controller do
```
Comment (svidlonix) [inline spec/controllers/comments_controller_spec.rb]: Please follow https://github.com/rubocop-hq/rspec-style-guide
when righting rspec

File: app/controllers/questions_controller.rb [hunk 6]
```diff
@@ -39,7 +49,12 @@ def update
   end
 
   def destroy
-    @question.answers.delete_all
+    @question.comments.destroy_all
```
[the remark this answers is not in the document]
Comment (ValAntG) [reply app/controllers/questions_controller.rb]: When will use polymorphic relationship, function "dependent: :destroy" don't work

File: app/forms/answer_form.rb (hunk 4 above)
Comment (ValAntG) [reply app/forms/answer_form.rb]: When I use answer_form, attachment and answer add to database after create  answer

File: Gemfile [hunk 7]
```diff
@@ -3,17 +3,22 @@ git_source(:github) { |repo| "https://github.com/#{repo}.git" }
...
 gem 'less-rails', git: 'https://github.com/MustafaZain/less-rails'
 gem 'pry', '~> 0.12.2'
 gem 'pundit'
 gem 'remotipart', github: 'jbox-web/remotipart', tag: '1.5.0'
 gem 'therubyracer'
+# gem 'thin'
 gem 'twitter-bootstrap-rails'
 gem 'virtus', '~> 1.0', '>= 1.0.5'
 # Bundle edge Rails instead: gem 'rails', github: 'rails/rails'
+gem 'rabl-rails'
```
[the remark this answers is not in the document]
Comment (svidlonix) [reply Gemfile]: Please reply.

File: app/assets/javascripts/answerLink.js [hunk 8]
```diff
@@ -0,0 +1,83 @@
+$(document).on("turbolinks:load", function () {
+  $('a.edit-answer-link').on('click ajax:success', toggleEditAnswerMode);
+  App.cable.subscriptions.create({channel: 'AnswersChannel', question_id: gon.question_id},
```
[the remark this answers is not in the document]
Comment (svidlonix) [reply app/assets/javascripts/answerLink.js]: Please fix it too.

File: app/assets/javascripts/answerLink.js [hunk 9]
```diff
@@ -0,0 +1,79 @@
+$(document).on("turbolinks:load", function () {
```
Comment (svidlonix) [inline app/assets/javascripts/answerLink.js]: Please choose one style way use ' or ".

File: app/assets/javascripts/answers.js [hunk 10]
```diff
@@ -1,42 +1,51 @@
...
-      $('.answer-errors').html(evt.detail[2].responseText);
-    } else {
-      errors = $.parseJSON(evt.detail[2].responseText);
-      $.each(errors, function (index, value) {
-        $('.answer-errors').append('<p>' + value + '</p>');
-      });
-    };
-  });
-})
+// $(document).on("turbolinks:load", function () {
```
[the remark this answers is not in the document]
Comment (svidlonix) [reply app/assets/javascripts/answers.js]: Why don't you delete this legacy code?

File: app/assets/javascripts/commentLink.js [hunk 11]
```diff
@@ -0,0 +1,63 @@
...
+    {
+      received(data) {
+        console.log(data)
+        if (data.action == 'create') { CommentCreate(data) };
+      }
+    });
+});
+
+function CommentCreate(data) {
+
```
[the remark this answers is not in the document]
Comment (svidlonix) [reply app/assets/javascripts/commentLink.js]: Please delete empty line

File: app/controllers/comments_controller.rb [hunk 12]
```diff
@@ -0,0 +1,60 @@
+class CommentsController < ApplicationController
+  before_action :load_elements, only: %i[update destroy]
+
+  def show; end
+
+  def create
+    @question = Question.find(params[:question_id]) if params[:question_id]
+    @answer = Answer.find(params[:answer_id]) if params[:answer_id]
+    @commentable = @answer || @question
```
Comment (svidlonix) [inline app/controllers/comments_controller.rb]: This functionality looks huge.

File: app/controllers/comments_controller.rb [hunk 13]
```diff
@@ -0,0 +1,60 @@
...
+      "questions/#{question_id}/comments", {
+        comment: CommentSerializer.new(comment), action: action, commentable_type: comment.commentable_type,
+        commentable_id: comment.commentable_id
+      }.as_json
+    )
+  end
+
+  def load_elements
+    @comment = Comment.find(params[:id])
+    @question = @comment.commentable
```
Comment (svidlonix) [inline app/controllers/comments_controller.rb]: This logic is realy unclear.

**Pull request merged**